WISE-Community/WISE-Client
# refactor(TeacherProjectService): Extract CreateComponentService

## Changes
- Extract functions related to creating a component from ```TeacherProjectService.createComponent()``` to ```CreateComponentService.create()```
- Refactor ```addComponentToNode()```
- Add tests
- Remove unused function ```TeacherProjectService.removeNodeIdFromInactiveNodes()```

## Test
- In the AT, creating new component(s) works as before. Verify that the new component is placed correctly.
   - Add your own node. Try adding multiple components
   - Edit step view. Try adding component at the beginning and after an existing component
